flake.nix

```diff
@@ -3,166 +3,190 @@
 
   inputs = {
     nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
     rust-overlay = {
       url = "github:oxalica/rust-overlay";
-      inputs.nixpkgs.follows = "nixpkgs";
+      inputs = {
+        nixpkgs.follows = "nixpkgs";
+        flake-utils.follows = "flake-utils";
+      };
     };
-    nci = {
-      url = "github:yusdacra/nix-cargo-integration";
-      inputs.nixpkgs.follows = "nixpkgs";
+    crane = {
```

Comment:
We could also remove crane and use only rust-overlay like [nil does](https://github.com/oxalica/nil/blob/main/flake.nix). Crane has some nice helpers for the `checks` and it seems to be very popular and stable though so I think it's fine to depend on it for now. If it becomes a problem in the future though, pure `rust-overlay` is an option